llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -435,6 +433,98 @@ def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
   }];
 }
 
+def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack", [
+  DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>,
+  DeclareOpInterfaceMethods<LinalgExtInterface>,
+  DeclareOpInterfaceMethods<TilingInterface,
+    ["getIterationDomain",
+     "getLoopIteratorTypes",
+     "generateScalarImplementation"]>,
+  DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
+]>{
+  let summary = "pack operation"; 
+  let description = [{
+    The pack operation takes a shaped type as input, tiles, and transposes it
```

Comment:
Nit:  I think it would be easier to say something like
```
The pack operation converts an `input` into a tiled and packed layout. The dimensions to be tiled are obtained from `dims_pos` and the size of the tile is obtained from `inner_tiles`. The dimensions listed in `dims_pos` need not be contiguous in which case the tile will get transposed.
```

Having some IR examples for different cases would also be useful.